docs/proposals/02-backupinfra.md

```diff
@@ -0,0 +1,232 @@
+    - Shoot controller will choose backup container as per the seed region. (With shoot control plane migration also, seed’s availibility zone might change but the region will be remaining same as per current scope)
+  - Otherwise cross provider case i.e. spec.backup.provider = spec.(type of provider)
+    - report validation error: Since we can’t expect `spec.backup.provider` = aws  to support region in, `spec.packet.constraint.region`.
+
+
+![sequence-diagram](Backup-Infrastructure-Provisioning-sequence-diagram.svg)
+
+#### Reconciliation
+
+Reconciliation on backup entry in seed cluster mostly comes in picture at the time of deletion. But we can add initialization steps like creation of [directory](#terminology) specific to shoot in backup bucket. We can simply create BackupEntry at the time of shoot deletion as well.
+
+#### Deletion
+- On shoot deletion, the BackupEntry instance i.e. shoot specific instance will get deletion timestamp because of ownerReference.
+- If `deletionGracePeriod` configured in GCM component configuration is expired, BackupInfrastructure Controller will delete the backup folder associated with it from backup object store.
+- Finally, it will remove the `finalizer` from backupEntry instance.
+
+### Alternative
+
+![sequence-diagram](Backup-Infrastructure-Provisioning-with-deletion-job.svg)
+
+## Discussion points / variations
+### Manual vs dynamic bucket creation
+- As per limit observed on different cloud providers, we can have single bucket for backups on one cloud providers. So, we could avoid the little complexity introduced in above approach by pre-provisioning buckets as a part of landscape setup. But there won't be anybody to detect bucket existence and its reconciliation.
+
+- Another thing we can have is, we can let administrator register the pool of root backup infra resource and let the controller schedule backup on one of this.
+
+- One more variation here could be to create bucket dynamically per hash of shoot UID.
+
+
+### SDK vs Terraform
+Initial reason for going for terraform script is its stability and the provided parallelism/concurrency in resource creation. For backup infrastructure, Terraform scripts are very minimal right now. Its simply have bucket creation script. With shared bucket logic, if possible we might want to isolate access at [directory](#terminology) level but again its additional one call. So, we will prefer switching to SDK for all object store operations.
+
+### Limiting the number of shoots per bucket
+We can have central map of configuration in `gardener-controller-component-configuration.yaml`.
+Where we will mark supported count of shoots per cloud provider. Most probable space could be,
```

Comment:
Can you make an example how that would look like?

Reply:
Updated in doc.